transformer.py

```diff
@@ -53,22 +58,36 @@ def construct_graph():
 			del user_subs['___last_seen']
 
 		for subreddit1 in user_subs:
-			get_or_add_node(node_dict, subreddit1)
-						
 			for subreddit2 in user_subs:
 				if subreddit1 is subreddit2:
 					continue
 
-				get_or_add_node(node_dict, subreddit2)
 				add_or_increment_edge(edge_dict, subreddit1, subreddit2)
 
+	edge_list = edge_dict.values()
+
+	if filter_min > 1:
+		print 'filtering results below ' + str(filter_min)
+		filtered = []
+		for edge in edge_list:
+			if edge['value'] >= filter_min:
+				filtered.append(edge)
+		edge_list = filtered
```

Comment:
I'm trying to estimate the memory footprint this approach has, since you're copying the values instead of removing them when they have a edge value below the filter. Do you think it's significant?